libvast/src/segment_store.cpp

```diff
@@ -394,16 +394,17 @@ caf::error
 segment_store::register_segment(const std::filesystem::path& filename) {
   auto chk = chunk::mmap(filename);
   if (!chk)
-    return caf::make_error(ec::filesystem_error, "failed to mmap chunk",
-                           filename.string());
+    return caf::make_error(ec::filesystem_error,
+                           fmt::format("failed to mmap chunk {}: {}", filename,
+                                       caf::to_string(chk.error().context())));
```

Comment:
It suffices to forward the error here.

```suggestion
    return std::move(chk.error());
```